pkg/syncthing/client.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 The envd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syncthing
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+func NewAPIClient() *http.Client {
+	return &http.Client{
+		Timeout: 60 * time.Second,
+	}
+}
+
+const (
+	GET  = "GET"
+	POST = "POST"
+	PUT  = "PUT"
+)
+
+type Client struct {
+	ApiKey   string
+	Client   *http.Client
+	BasePath string
+}
+
+func (s *Syncthing) NewClient() *Client {
+	return &Client{
+		ApiKey:   s.ApiKey,
+		Client:   NewAPIClient(),
+		BasePath: fmt.Sprintf("http://localhost:%s", s.Port),
```

Comment:
localhost can be resolved to an IPv6 address. Does this tool support that? Or you just mean the loopback address.